app/modules/transcribe/overlay/keypad-slide.directive.js

```diff
@@ -48,6 +47,25 @@ function keypadSlide(hotkeys, overlayConfig) {
             }
             textarea.caret(startTag.length + text.length);
         }
+
+        function preLoad() {
+            var deferred = $q.defer();
+            var imageArray = [];
+            for (var i = 0; i < $scope.abbreviations.length; i++) {
+                imageArray[i] = new Image();
+                imageArray[i].src = $scope.abbreviations[i].imgPath;
+            }
+            imageArray.forEach.onload = function () {
```

Comment:
No, there's only one promise. `forEach` will run across all of them, but will resolve the promise as soon as one is loaded. What you need is an array of promises, one for each image, and for the whole thing to resolve as soon as they're _all_ done.

Also, I can tell it's not working as there's a noticeable lag between mousing over some buttons and getting the popover. Try doing a hard refresh so they're not locally cached and you'll see.